### PR TITLE
feat: setting welcomePost as showOnFeed as false by default

### PR DESCRIPTION
**Pull request by TomDowling**
Setting the showOnFeed flag as false by default for welcomePosts

### CODE REVIEW

**Review by rebelchris — CHANGES_REQUESTED** [review]
this is great, part of this ticket was also to retro fix existing squads (which will need a script probably)
Would you want to take that one on as well? (else let me know I can support on it)

**Comment by idoshamun** [comment]
@rebelchris I think a simple migration can also be useful as a retro fix. Should be straightforward

File: src/common/post.ts [hunk 1]
```diff
@@ -145,6 +145,7 @@ export const createSquadWelcomePost = async (
       banned: true,
       private: true,
       visible: true,
+      showOnFeed: false,
```
Comment (idoshamun) [inline src/common/post.ts:148]: Currently `flags` is not used as a primary source so we should add `showOnFeed` as a main property of the object

**Review by idoshamun — CHANGES_REQUESTED** [review]
Please check my comment. Also we should update the tests I guess right?

**Comment by rebelchris** [comment]
@idoshamun Added all comments and a new test case as we didn't explicitly test for it

File: src/migration/1689854830946-WelcomeShowOnFeed.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class WelcomeShowOnFeed1689854830946 implements MigrationInterface {
+  name = 'WelcomeShowOnFeed1689854830946';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `UPDATE post SET flags = flags || '{"showOnFeed": false}' WHERE type = 'welcome'`,
```
Comment (idoshamun) [inline src/migration/1689854830946-WelcomeShowOnFeed.ts:8]: We need to set `showOnFeed` itself as well here

Comment (idoshamun) [reply src/migration/1689854830946-WelcomeShowOnFeed.ts:8]: I see it only in `down` the `up` wasn't changed

**Comment by rebelchris** [comment]
@idoshamun rushing was a bad idea, fixed both now

**Review by idoshamun — APPROVED** [review]
Perfect!

**Review by sshanzel — APPROVED** [review]
Looks great!

**Pull request merged**